framework/PostProcessors/CrossValidation.py

```diff
@@ -306,45 +295,40 @@ def run(self, inputIn):
         break
     if cvEngine is None:
       self.raiseAnError(IOError, "No cross validation engine is provided!")
+
     outputDict = {}
+    # construct matrix and pass matrix
     for trainIndex, testIndex in cvEngine.generateTrainTestIndices():
       trainDict, testDict = self.__generateTrainTestInputs(inputDict, trainIndex, testIndex)
       ## Train the rom
       cvEstimator.train(trainDict)
       ## evaluate the rom
       outputEvaluation = cvEstimator.evaluate(testDict)
-      ## Compute the distance between ROM and given data using Metric system
-      for targetName, targetValue in outputEvaluation.items():
-        if targetName not in outputDict.keys():
-          outputDict[targetName] = {}
-        for metricInstance in self.metricsDict.values():
-          metricValue = metricInstance.distance(targetValue, testDict[targetName])
-          if hasattr(metricInstance, 'metricType'):
-            if metricInstance.metricType not in self.validMetrics:
-              self.raiseAnError(IOError, "The metric type: ", metricInstance.metricType, " can not be used, the accepted metric types are: ", str(self.validMetrics))
-            metricName = metricInstance.metricType
-          else:
-            metricName = metricInstance.type
-          metricName = metricInstance.name + '_' + metricName
-          if metricName not in outputDict[targetName].keys():
-            outputDict[targetName][metricName] = []
-          outputDict[targetName][metricName].append(metricValue[0])
+      cvOutputs = np.asarray(outputEvaluation.values()).T
+      testOutputs = np.asarray([testDict[var] for var in outputEvaluation.keys()]).T
+      for metricInstance in self.metricsDict.values():
+        metricValue = metricInstance.distance(cvOutputs, testOutputs)
+        if hasattr(metricInstance, 'metricType'):
+          if metricInstance.metricType not in self.validMetrics:
+            self.raiseAnError(IOError, "The metric type: ", metricInstance.metricType, " can not be used, the accepted metric types are: ", str(self.validMetrics))
+        else:
+            self.raiseAnError(IOError, "The metric: ", metricInstance.name, " can not be used, the accepted metric types are: ", str(self.validMetrics))
+
+        varName = 'cv' + '_' + metricInstance.name + '_' + cvEstimator.name
+        if varName not in outputDict.keys():
+          outputDict[varName] = []
+        outputDict[varName].append(metricValue[0])
+
     scoreDict = {}
-    if not self.cvScores:
-      return outputDict
```

Comment:
So you removed the possibility to get the scores for each fold?

Reply:
Can you put the "update documenation" in the todolist for this work? (The shared sheet in google doc)

Reply:
Yes, I think this PP should return a single quantity to tell the user if the ROM is good or not.

Reply:
I have updated the shared excel sheet.

Reply:
I agree it should by default, but honestly I can use the cross validation (k-fold) also as metric to understand how homogeneously my regression is able to capture the underlying model. I would have changed the default (to be the single scalar) but not remove the possibility to get the stats from all the folds...